packages/flame/lib/src/effects/sequence_effect.dart

```diff
@@ -22,6 +22,32 @@ EffectController _createController({
   return ec;
 }
 
+/// Specifies how to alternate a [SequenceEffect] pattern.
+///
+/// If [AlternatePattern.none] is provided, then
+/// the [SequenceEffect] does not repeat in reverse when the child
+/// [Effect]s have all completed playing.
+///
+/// When [SequenceEffect] is provided a pattern other than
+/// [AlternatePattern.none], the sequence will repeat in the
+/// reverse order, doubling the length of [EffectController.duration].
+///
+/// [AlternatePattern.includeLast] will replay the last [Effect] at the start
+/// of the alternate pattern, effectively playing it twice in a row.
+///
+/// [AlternatePattern.excludeLast] will not replay the last [Effect] and will
+/// jump to the second-to-last [Effect], if available, at the start of the
+/// alternate pattern instead, effectively playing the last [Effect] only once
+/// throughout the original and alternating pattern.
+enum AlternatePattern {
+  none(0),
+  includeLast(1),
+  excludeLast(2);
```

Comment:
If `inifinite` (or N runs) is used together with `alternate`, shouldn't the first also be excluded on runs that are after the first and not the last?

Reply:
I've corrected the default pattern behavior. The alternating pattern now plays both the first and last `Effect` elements in the list, instead of just the last element as it did before this PR. This default behavior is `AlternatePattern.repeatLast`. The term "last" used here is relative to the playback direction.

To omit placing first and last twice-in-a-row when the sequence reverses, `AlternatePattern.doNotRepeatLast` is appropriate.